### PR TITLE
Allow setting Guzzle options on a service

### PR DESCRIPTION
**Pull request by Sazpaimon**
See #19 

/cc @nowenL

### CODE REVIEW

**Comment by Sazpaimon** [comment]
I opted to use `http` instead of `guzzle` for the options key in case the underlying HTTP implementation changes.

File: src/Common/Internal/ServiceRestProxy.php [hunk 1]
```diff
@@ -50,78 +50,86 @@ class ServiceRestProxy extends RestProxy
...
+     *
      * @var \Uri
      */
        private $_psrUri;
-    
+
+    /**
+     * @var array
+     */
+    private $options;
```
Comment (nowenL) [inline src/Common/Internal/ServiceRestProxy.php]: use $_options to keep align with the current practice.

**Comment by nowenL** [comment]
Thx! Add one minor comment. Pls help to address. We will accept the PR after validation.

**Comment by Sazpaimon** [comment]
Done!

**Comment by Sazpaimon** [comment]
@nowenL ping?

**Comment by nowenL** [comment]
Hi, Sazpaimon. Sorry for the late reply. 

The validation is done. However, before we can accept your PR, you need to sign a contribution license agreement. Since your PR is the first one we are taking from community, the process is new to us. I'll follow this up with our CLA guys and reply you by Monday. Sorry for the inconvenience.

**Comment by Sazpaimon** [comment]
@nowenL I have already signed the CLA and have had multiple PRs accepted on the Azure PHP SDK already (Azure/azure-sdk-for-php#809 Azure/azure-sdk-for-php#764 Azure/azure-sdk-for-php#748) so I don't expect that to be an issue.

**Comment by nowenL** [comment]
Oh. That's great. Thanks for you contribution.

@vinjiang, pls accept this PR.

**Comment by Sazpaimon** [comment]
Thanks, now I just need 0.10.2 to get tagged so I can use this in my project.

**Comment by Sazpaimon** [comment]
@nowenL @vinjiang any ETA on when this will get put into a stable release? Don't want to add a dependency to a dev version unless I have to.

**Comment by nowenL** [comment]
@Sazpaimon, we are having discussion about having a release with this change. Will come back ASAP once we have an conclusion. Thx

**Comment by nowenL** [comment]
@Sazpaimon. A small version with your pr and some bug fix will be release in the week of 8.15th.

**Pull request merged**